core/templates/pages/topic-editor-page/modal-templates/create-new-subtopic-modal.component.ts

```diff
@@ -91,8 +91,11 @@ export class CreateNewSubtopicModalComponent
     this.subtopicTitle = '';
     this.errorMsg = null;
     this.subtopicUrlFragmentExists = false;
-    this.topicUpdateService
-      .addSubtopic(this.topic, this.subtopicTitle, this.editableUrlFragment);
+
+    setTimeout(() => {
+      this.topicUpdateService
+        .addSubtopic(this.topic, this.subtopicTitle, this.editableUrlFragment);
+    });
```

Comment:
if we are not using setTimeout then the ngOnint adds a new subtopic in the list on the go and as a new subtopic is added the warning count also increases from 3 to 5 (2 new subtopic warnings as shown in the image below). 
![Screenshot from 2022-12-20 01-18-47](https://user-images.githubusercontent.com/77006897/208508319-d2a50816-d19d-4193-ab7c-c6ddeb25ca31.png)

I don't know why but the set timeout solves the problem, (if we are using set timeout then the warning count is not increased) (just for info: This is only a console error it will not affect the user).

Reply:
I think this seems weird behaviour, right? The ngOnInit should not even add a subtopic to the list until the user has clicked the "confirm" button on the modal.

Can't we just store the subtopic dict locally in the component and only add it to the list when the user confirms addition of the subtopic?

Reply:
I tried doing this thing, doing this makes the modal displaying nothing, as we have used *ngIf on the modal template file that is dependent on this.topicUpdateService.addSubtopic(this.topic, this.subtopicTitle, this.editableUrlFragment).

Reply:
@seanlip PTAL

Reply:
i have updated the logic for creating subtopic everything seems working now.